File: src/Csharp/YandexDiskSDK/YandexDiskSDK/JSON.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace YandexDiskSDK.JSON
{

    public class JSON_Error
    {
        [JsonProperty("description")]public string _ErrorMessage;
        [JsonProperty("error")] public int _ErrorCode;
    }

    #region JSONexchangingVerificationCodeForToken
    public class JSONexchangingVerificationCodeForToken
    {
        public string token_type { get; set; }
        public string access_token { get; set; }
        public int expires_in { get; set; }
        public string refresh_token { get; set; }
    }
    #endregion

    #region JSON_CheckOperationStatus
    public class JSON_CheckOperationStatus
    {
        [Browsable(false)]
        [Bindable(false)]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        [EditorBrowsable(EditorBrowsableState.Never)]
        public string status { get; set; }

        public JSON_CheckOperationStatus.OperationStatus ResultStatus
        {
            get
            {
                OperationStatus rslt = OperationStatus.success;
                switch (this.status)
                {
                    case "success":
                        rslt = OperationStatus.success;
                        break;
                    case "failure":
                        rslt = OperationStatus.failure;
                        break;
                    case "in-progress":
                        rslt = OperationStatus.inprogress;
                        break;
                    default:
                        rslt = OperationStatus.error;
                        break;
                }
                return rslt;
            }
        }

        public enum OperationStatus
        {
            success,
            failure,
            inprogress,
            error
        }
    }
    #endregion


    #region JSON_UserInfo
    public class JSON_UserInfo
    {
        public string max_file_size { get; set; }
        public bool unlimited_autoupload_enabled { get; set; }
        public string total_space { get; set; }
        public string trash_size { get; set; }
        public bool is_paid { get; set; }
        public string used_space { get; set; }
        public JSONSystem_Folders system_folders { get; set; }
        public JSON_User user { get; set; }
        public string revision { get; set; }
    }
    public class JSONSystem_Folders
    {
        public string odnoklassniki { get; set; }
        public string google { get; set; }
        public string instagram { get; set; }
        public string vkontakte { get; set; }
        public string mailru { get; set; }
        public string downloads { get; set; }
        public string applications { get; set; }
        public string facebook { get; set; }
        public string social { get; set; }
        public string screenshots { get; set; }
        public string photostream { get; set; }
    }
    #endregion

    #region JSON_FolderList
    public class JSON_FolderList
    {
        public List<JSON_FileMetadata> _Files { get; set; }
        public List<JSON_FolderMetadata> _Folders { get; set; }
        public string public_key { get; set; }
        public string sort { get; set; }
        public int limit { get; set; }
        public int offset { get; set; }
        public string path { get; set; }
        public int total { get; set; }
        public bool HasMore
        {
            get
            {
                return this.total >= checked(this.limit + this.offset);
            }
        }
    }
    #endregion

    #region JSON_User
    public class JSON_User
    {
        public string country { get; set; }
        public string login { get; set; }
        public string display_name { get; set; }
        public string uid { get; set; }
    }
    #endregion

    #region JSON_FilesList
    public class JSON_FilesList
    {
        [JsonProperty("items")]
        public List<JSON_FileMetadata> _Files { get; set; }
        public int limit { get; set; }
        public int offset { get; set; }
    }
    #endregion

    #region JSON_FileMetadata
    public class JSON_FileMetadata
    {
        public string name { get; set; }
        public Exif exif { get; set; }
        public DateTime created { get; set; }
        public DateTime modified { get; set; }
        public string path { get; set; }
        public utilitiez.ItemTypeEnum type { get; set; }
        public int size { get; set; }
        public string mime_type { get; set; }
        public string media_type { get; set; }
        public string preview { get; set; }
        public string sha256 { get; set; }
        public string md5 { get; set; }
        public string public_key { get; set; }
        public string public_url { get; set; }
        [JsonProperty("file")]public string DownloadUrl { get; set; }
        public bool IsShared{get{return string.IsNullOrEmpty(this.public_key);}}
    }

    public class Exif
    {
        public DateTime date_time { get; set; }
    }


    #endregion

    #region JSON_FolderMetadata
    public class JSON_FolderMetadata
    {
        public DateTime modified { get; set; }
        public string name { get; set; }
        public DateTime created { get; set; }
        public string path { get; set; }
        public utilitiez.ItemTypeEnum type { get; set; }
        public string public_key { get; set; }
        public string public_url { get; set; }
        public bool IsShared{get{return string.IsNullOrEmpty(this.public_key);}}
    }
    #endregion

    #region JSON_MixedMetadata
    public class JSON_MixedMetadata
    {
        public string name { get; set; }
        public Exif exif { get; set; }
        public DateTime created { get; set; }
        public DateTime modified { get; set; }
        public string path { get; set; }
        public utilitiez.ItemTypeEnum type { get; set; }
        public int size { get; set; }
        public string mime_type { get; set; }
        public string media_type { get; set; }
        public string preview { get; set; }
        public string sha256 { get; set; }
        public string md5 { get; set; }
        public string public_key { get; set; }
        public string public_url { get; set; }
       [JsonProperty("file")]
        public string DownloadUrl { get; set; }
        public bool IsShared{get{return string.IsNullOrEmpty(this.public_key);}}
    }
    #endregion

    #region JSON_PublicFolder
    public class JSON_PublicFolder
    {
        public string public_key { get; set; }
        public string public_url { get; set; }
        [JsonProperty("_embedded")]public JSON_FolderList ItemsList { get; set; }
        public string name { get; set; }
        public DateTime created { get; set; }
        public DateTime modified { get; set; }
        public JSON_User owner { get; set; }
        public string path { get; set; }
        public utilitiez.ItemTypeEnum type { get; set; }
        public int views_count { get; set; }
    }
    #endregion




}

File: src/VB/YandexDiskSDK/YandexDiskSDK/JSON.vb
Imports System.ComponentModel
Imports Newtonsoft.Json
Imports YandexDiskSDK.utilitiez

Namespace JSON

    Public Class JSON_Error
        <JsonProperty("description")> Public Property _ErrorMessage As String
        <JsonProperty("error")> Public Property _ErrorCode As String
    End Class

#Region "CheckOperationStatus"
    Public Class JSON_CheckOperationStatus
        Enum OperationStatus
            success
            failure
            inprogress
            [error]
        End Enum
        <Browsable(False)> <Bindable(False)> <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> <EditorBrowsable(EditorBrowsableState.Never)>
        Public Property status As String
        Public ReadOnly Property ResultStatus As OperationStatus
            Get
                Dim rslt As New OperationStatus
                If status = "success" Then
                    rslt = OperationStatus.success
                ElseIf status = "failure" Then
                    rslt = OperationStatus.failure
                ElseIf status = "in-progress" Then
                    rslt = OperationStatus.inprogress
                Else
                    rslt = OperationStatus.error
                End If
                Return rslt
            End Get
        End Property

    End Class
#End Region



#Region "userinfo"
    Public Class JSON_UserInfo
        Public Property max_file_size As String
        Public Property unlimited_autoupload_enabled As Boolean
        Public Property total_space As String
        Public Property trash_size As String
        Public Property is_paid As Boolean
        Public Property used_space As String
        Public Property system_folders As JSONSystem_Folders
        Public Property user As JSON_User
        Public Property revision As String
    End Class
    Public Class JSONSystem_Folders
        Public Property odnoklassniki As String
        Public Property google As String
        Public Property instagram As String
        Public Property vkontakte As String
        Public Property mailru As String
        Public Property downloads As String
        Public Property applications As String
        Public Property facebook As String
        Public Property social As String
        Public Property screenshots As String
        Public Property photostream As String
    End Class
#End Region



    Public Class JSON_PublicFolder
        Public Property public_key As String
        Public Property public_url As String
        <JsonProperty("_embedded")> Public Property ItemsList As JSON_FolderList
        Public Property name As String
        Public Property created As Date
        Public Property modified As Date
        Public Property owner As JSON_User
        Public Property path As String
        Public Property type As ItemTypeEnum
        Public Property views_count As Integer
    End Class

    Public Class JSON_User
        Public Property country As String
        Public Property login As String
        Public Property display_name As String
        Public Property uid As String
    End Class

    Public Class JSON_FolderList
        Public Property _Files As List(Of JSON_FileMetadata)
        Public Property _Folders As List(Of JSON_FolderMetadata)

        Public Property public_key As String
        Public Property sort As String
        Public Property limit As Integer
        Public Property offset As Integer
        Public Property path As String
        Public Property total As Integer
        Public ReadOnly Property HasMore As Boolean
            Get
                Return If(total < limit + offset, False, True)
            End Get
        End Property
    End Class

    Public Class JSON_FilesList
        <JsonProperty("items")> Public Property _Files As List(Of JSON_FileMetadata)
        Public Property limit As Integer
        Public Property offset As Integer
    End Class

    Public Class JSON_FoldersList
        <JsonProperty("items")> Public Property _Folders As List(Of JSON_FolderMetadata)
        Public Property limit As Integer
        Public Property offset As Integer
    End Class

    Public Class JSON_MixedMetadata
        Public Property name As String
        Public Property exif As Exif
        Public Property created As Date
        Public Property modified As Date
        Public Property path As String
        Public Property type As ItemTypeEnum
        Public Property size As Integer
        Public Property mime_type As String
        Public Property media_type As String
        Public Property preview As String
        Public Property sha256 As String
        Public Property md5 As String
        Public Property public_key As String
        Public Property public_url As String
        <JsonProperty("file")> Public Property DownloadUrl As String
        Public ReadOnly Property IsShared As Boolean
            Get
                Return String.IsNullOrEmpty(public_key)
            End Get
        End Property
    End Class

    Public Class JSON_FileMetadata
        Public Property name As String
        Public Property exif As Exif
        Public Property created As Date
        Public Property modified As Date
        Public Property path As String
        Public Property type As ItemTypeEnum
        Public Property size As Integer
        Public Property mime_type As String
        Public Property media_type As String
        Public Property preview As String
        Public Property sha256 As String
        Public Property md5 As String
        Public Property public_key As String
        Public Property public_url As String
        <JsonProperty("file")> Public Property DownloadUrl As String
        Public ReadOnly Property IsShared As Boolean
            Get
                Return String.IsNullOrEmpty(public_key)
            End Get
        End Property
    End Class

    Public Class JSON_FolderMetadata
        Public Property modified As Date
        Public Property name As String
        Public Property created As Date
        Public Property path As String
        Public Property type As ItemTypeEnum
        Public Property public_key As String
        Public Property public_url As String
        Public ReadOnly Property IsShared As Boolean
            Get
                Return String.IsNullOrEmpty(public_key)
            End Get
        End Property
    End Class

    Public Class Exif
        Public Property date_time As Date
    End Class

    Public Class Comment_Ids
        Public Property private_resource As String
        Public Property public_resource As String
    End Class

#Region "token"
    Public Class JSONexchangingVerificationCodeForToken
        Public Property token_type As String
        Public Property access_token As String
        Public Property expires_in As Integer
        Public Property refresh_token As String
    End Class
#End Region




End Namespace

